backend/app/api/endpoints/adapter/attachments.py

```
@@ -176,13 +176,20 @@ async def download_attachment(
     )
     if attachment is None:
         raise HTTPException(status_code=404, detail="Attachment not found")
-    
+
+    # Get binary data from appropriate storage (MinIO or MySQL)
+    try:
+        binary_data = attachment_service.get_attachment_binary(attachment)
+    except ValueError as e:
+        logger.error(f"Failed to retrieve attachment binary data: {e}")
```

Ruff (0.14.7)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)